nar/reader.go

```diff
@@ -20,9 +20,6 @@ const (
 	// maximum length for a single path element
 	// NAME_MAX is 255 on Linux
 	nameLenMax = 255
-	// maximum length for a relative path
-	// PATH_MAX is 4096 on Linux, but that includes a null byte
-	pathLenMax = 4096 - 1
```

Comment:
Did we miss checking for this in the reader? We should probably re-introduce the check, instead of just dropping the (unused) constant.

Reply:
We did miss checking this, and checked the wrong thing for symlink targets. Pushed fixes in a834a0e01fc5bef5375085ef24b29a5798f4fab1 and b2adcab0fa8b4a7984d64d02fd38ad76ad7cc7b4.